panther_driver/src/driver_node.py

```diff
@@ -1,323 +1,361 @@
-        pose_msg = Pose()
-        pose_msg.position.x = 0
-        pose_msg.position.y = 0
-        pose_msg.position.z = 0
-        pose_msg.orientation.x = 0
-        pose_msg.orientation.y = 0
-        pose_msg.orientation.z = 0
-        pose_msg.orientation.w = 1
-
-    if publish_odometry == True:
-        odom_publisher = rospy.Publisher("odom/wheel", Odometry, queue_size=1)
-        odom_msg = Odometry()
-
-    rospy.Subscriber("/cmd_vel", Twist, RK.cmdVelCallback, queue_size=1)
-
-    can_interface = rospy.get_param("~can_interface", "panther_can")
-    RK.robot_width = rospy.get_param("~robot_width", 0.682)
-    RK.robot_length = rospy.get_param("~robot_length", 0.44)
-    RK.wheel_radius = rospy.get_param("~wheel_radius", 0.1825)
-    RK.encoder_resolution = rospy.get_param("~encoder_resolution", 400*4)
-    RK.gear_ratio = rospy.get_param("~gear_ratio", 30.08)
-    RK.power_factor = rospy.get_param("~power_factor", 0.04166667)
-
-    if rospy.has_param("~eds_file"):
-        eds_file = rospy.get_param("~eds_file")
-    else:
-        rospy.logerr(f"[{rospy.get_name()}] eds_file not defined, can not start CAN interface")
-        return
-
-    loop_rate = 20
-    rate = rospy.Rate(loop_rate)
-    rospy.loginfo(f"[{rospy.get_name()}] Start with creating a network representing one CAN bus")
-    network = canopen.Network()
-    rospy.loginfo(f"[{rospy.get_name()}] Add some nodes with corresponding Object Dictionaries")
-    front_controller = canopen.RemoteNode(1, eds_file)
-    rear_controller = canopen.RemoteNode(2, eds_file)
-    network.add_node(front_controller)
-    network.add_node(rear_controller)
-    rospy.loginfo(f"[{rospy.get_name()}] Connect to the CAN bus")
-    network.connect(channel=can_interface, bustype="socketcan")
-
-    # rospy.loginfo(f"[{rospy.get_name()}] Reset encoders")
-    # front_controller.sdo["Cmd_SENCNTR"][1].raw = 0
-    # front_controller.sdo["Cmd_SENCNTR"][2].raw = 0
-
-    robot_x_pos = 0.0
-    robot_y_pos = 0.0
-    robot_th_pos = 0.0
-
-    wheel_pos = [0.0, 0.0, 0.0, 0.0]
-    wheel_vel = [0.0, 0.0, 0.0, 0.0]
-    wheel_curr = [0.0, 0.0, 0.0, 0.0]
-
-    last_time = rospy.Time.now()
-
-    # VARIABLE FOR ERROR TRACKING
-    err_count = 0
-
-    while not rospy.is_shutdown():
+from nav_msgs.msg import Odometry
+from std_srvs.srv import Trigger
+
+from panther_msgs.msg import DriverStateArr, FaultFlag, RuntimeError
+
+from panther_kinematics import PantherDifferential, PantherMecanum
+from panther_can import PantherCAN
+
+
+class PantherDriver(object):
+    def __init__(self, name) -> None:
+        rospy.init_node(name, anonymous=False)
+
+        assert rospy.has_param(
+            '~eds_file'
+        ), f'[{rospy.get_name()}] eds_file not defined, can not start CAN interface'
+        self._eds_file = rospy.get_param('~eds_file')
+        self._can_interface = rospy.get_param('~can_interface', 'panther_can')
+        self._kinematics_type = rospy.get_param('~wheel_type', 'differential')
```

Comment:
```suggestion
        self._kinematics_type = rospy.get_param('~kinematics_type', 'differential')
```
I think param name should be `kinematics_type`, this also require change in launch file